data/calculated_parser/functions.py

```diff
@@ -69,11 +70,293 @@ def sspeed(depth, latitude, temperature, salinity):
         press = [seawater.pres(d, latitude) for d in depth]
     except TypeError:
         press = seawater.pres(depth, latitude)
-
+    
     speed = seawater.svel(salinity, temperature, press)
     return np.array(speed)
 
 
+def sspeedmin(depth, lat, lon, temperature, salinity):
+    """
+    Finds the global minimum of the speed of sound
+
+    Parameters:
+    sspeed: Speed of Sound
+    latitude: The latitude(s) in degrees North
+    """
+    
+    speed = sspeed(depth, lat, temperature, salinity)
+    speed = speed.transpose()
+    
+    for x in range(speed.shape[0]):
+        for y in range(speed.shape[1]):
+            min_val = np.nanmin(speed[x,y])
+            idx = np.where(speed[x,y] == min_val)
+            if (np.isnan(min_val)):
+                speed[x,y] = np.nan
+            elif idx[0].shape[0] > 1:
+                idx = idx[0][0]
+
+                speed[x,y] = depth.values[idx]
+            else:
+                speed[x,y] = depth.values[idx]
+            #speed[x][y] = depth[np.where(speed[x][y] == np.nanmin(speed[x][y]))]  #np.nanmin(speed[x][y])
```

Comment:
I wonder if lines 88 to 108 can be replaced with
```python
minspeed = np.nanmin(speed, axis=0)

return minspeed
```
https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanmin.html
* If we use this, we are assuming the speed array always has depth as the first dimension. 
* If speed has dimensions [z,y,x]  then minspeed should have dimensions [y, x]
* This would give a runtime warning if a position has nans over the whole depth and then return a nan for that position

Reply:
This functions does something different than what the doscstrings  describe. It finds the depth of the minimum speed of sound (not the minimum speed of sound)

So ignore my comment above! I would suggest a more appropriate function name and description.

`np.nanargmin` might be useful here
https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanargmin.html

Reply:
Changed function name to soundchannelaxis()